src/cljs/athens/components/components_util.cljs

```diff
@@ -0,0 +1,25 @@
+(ns athens.components.components-util
+  (:require
+    [athens.components.todo :as todo]
+    [athens.components.website-embeds :as website-embeds]))
+
+
+(defn default-component
+  [content uid]
+  ((constantly nil) uid)
+  [:button content])
+
+;; TODO: use metaprogramming to achieve dynamic rendering with both basic components and custom components
+(defn render-component
+  "Renders a component using its parse tree & its uid."
+  [content uid]
+  (let [components (concat todo/components website-embeds/components)
```

Comment:
Because you've defined `components` across several namespaces, you need to join them at some point. That's not inherently a bad thing, but you are joining them every single time you call `render-component`, when the computed collection never actually changes.

Reply:
Makes sense - should've concated them only once